### ISSUE TITLE
import/no-self-import incorrectly identifies self.

### ISSUE THREAD

**Issue by nomad3k** [benmosher/eslint-plugin-import#1247]
The issue seems to be when you have a local directory that has the same name as a global import.  For example, I have a project that uses redis, and the code that uses it sits in a reds directory.

e.g. File: `redis/index.js`
```
const redis = require('redis'); // <-- import/no-self-import error here
```

**Comment by vkomodey** [comment]
The same situation: I have a file with name `./validator.js` where I require NPM `validator` library. So eslint produce the `import/no-self-import` error

**Comment by mikestopcontinues** [comment]
I know +1s are bad form, but this issue hasn't been touched in a while, so... +1 ;)

**Comment by ljharb** [comment]
PRs are welcome.

**Comment by gajus** [comment]
Same with matching file names.

A `turbowatch.ts` that imports `turbowatch` is also triggering this error.

```ts
import { watch } from 'turbowatch';

export default watch({
  project: __dirname,
  triggers: [],
});
```
